src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -80,9 +80,16 @@ class PathManager : IPathManager
         private readonly HashSet<string> preloadedLibraries;
         private readonly HashSet<string> extensionsDirectories;
         private readonly HashSet<string> viewExtensionsDirectories;
-        
+
         #endregion
 
+        internal IPreferences Preferences { get; set; }
+
+        private List<string> RootDirectories
```

Comment:
Can we get rid of rootDirectories? and change the return type to IEnumerable<string>?